src/neuroconv/datainterfaces/ecephys/spikeglx/spikeglxnidqinterface.py

```diff
@@ -0,0 +1,144 @@
+        metadata_schema = super().get_metadata_schema()
+        metadata_schema["properties"]["Ecephys"]["properties"].update(
+            ElectricalSeriesRaw=get_schema_from_hdmf_class(ElectricalSeries)
+        )
+        return metadata_schema
+
+    def get_metadata(self):
+        metadata = super().get_metadata()
+        session_start_time = get_session_start_time(self.meta)
+        if session_start_time:
+            metadata = dict_deep_update(metadata, dict(NWBFile=dict(session_start_time=session_start_time)))
+
+        # Device metadata
+        metadata = dict(Ecephys=dict())
+        device_key = "niDev1ProductName"  # At least, this is true for the GIN test data
+        if device_key in self.meta:
+            header_metadata = dict(self.meta)
+            for exclude_key in ["fileCreateTime", device_key]:
+                header_metadata.pop(exclude_key)
+            device = dict(
+                name=self.meta[device_key],
+                description=json.dumps(header_metadata),
+                manufacturer="National Instruments",
+            )
+            metadata["Ecephys"].get("Device", list()).append(Device=device)
+
+        # Add groups metadata
+        group_names = self.recording_extractor.get_property("group_name")
+        if self.recording_extractor.get_property("group_name") is None:
+            electrode_groups = [
+                dict(
+                    name="NIDQChannelGroup",
+                    description="A group representing the NIDQ channels.",
+                    location="unknown",
+                    device=device["name"],
+                )
+            ]
+        else:
+            electrode_groups = [
+                dict(
+                    name=group_name,
+                    description="A group representing the NIDQ channels.",
+                    location="unknown",
+                    device=device["name"],
+                )
+                for group_name in set(group_names)
+            ]
+        metadata["Ecephys"].get("ElectrodeGroup", list()).extend(electrode_groups)
+
+        # Electrodes columns descriptions
+        metadata["Ecephys"]["Electrodes"] = [
+            dict(name="group_name", description="Name of the ElectrodeGroup this electrode is a part of."),
+        ]
+
+        metadata["Ecephys"]["ElectricalSeriesNIDQ"] = dict(
+            name="ElectricalSeriesNIDQ", description="Raw acquisition traces from the NIDQ (.nidq.bin) channels."
+        )
+        return metadata
+
+    def get_channel_names(self) -> List[str]:
+        """Return a list of channel names as set in the recording extractor."""
+        return list(self.recording_extractor.get_channel_ids())
+
+    def get_event_starting_times_from_ttl(self, channel_name: str) -> np.ndarray:
+        """
+        Return the start of event times from the rising part of TTL pulses on one of the NIDQ channels.
+
+        Parameters
+        ----------
+        channel_name : int
```

Comment:
```suggestion
        channel_name : str
```

Reply:
I'm not sure whether it should be an int or a str, but if it's an int then it shouldn't be called "name", and if it's a str then the docstring should indicate that

Reply:
Should be `str`